azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineScaleSetApiMockTest.java

```diff
@@ -0,0 +1,271 @@
+//import org.jclouds.azurecompute.arm.domain.OSProfile.WindowsConfiguration.WinRM.Protocol;
+//import org.jclouds.azurecompute.arm.domain.Secrets.SourceVault;
+import org.jclouds.azurecompute.arm.domain.StorageProfile;
+import org.jclouds.azurecompute.arm.domain.Subnet;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSet;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetDNSSettings;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetIpConfiguration;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetIpConfigurationProperties;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetNetworkProfile;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetNetworkSecurityGroup;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetOSProfile;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetPlan;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetProperties;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetPublicIPAddressConfiguration;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetPublicIPAddressProperties;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetSKU;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetUpgradePolicy;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineScaleSetVirtualMachineProfile;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiMockTest;
+import org.testng.annotations.Test;
+
+//import java.net.URI;
+//import java.text.DateFormat;
+//import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+//import static com.google.common.collect.Iterables.isEmpty;
+//import static org.jclouds.util.Predicates2.retry;
+//import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+
+@Test(groups = "unit", testName = "VirtualMachineApiMockTest", singleThreaded = true)
+public class VirtualMachineScaleSetApiMockTest extends BaseAzureComputeApiMockTest {
+
+    private final String subscriptionid = "SUBSCRIPTIONID";
+    private final String resourcegroup = "myresourcegroup";
+    private final String virtualNetwork = "myvirtualnetwork";
+    private final String subnetName = "mysubnet";
+    private final String apiVersion = "2016-04-30-preview";
+
+
+    private VirtualMachineScaleSet getVMSS(VirtualMachineScaleSetPlan plan) {
+        VirtualMachineScaleSet vmss = VirtualMachineScaleSet.create("/subscriptions/SUBSCRIPTIONID/" + ""
+                        + "resourceGroups/groupname/providers/Microsoft.Compute/VirtualMachineScaleSets/virtualmachinescaleset", "windowsmachine",
+                "eastus", getSKU(), null, plan, getVirtualMachineScaleSetProperties());
+        return vmss;
+    }
+
+
+
+    public VirtualMachineScaleSetSKU getSKU() {
```

Comment:
All these methods are duplicated. Make them visible in the other class and just reuse them.

Also, this class has no tests. Add the corresponding mock tests.